kusto/client_details.go

```diff
@@ -0,0 +1,131 @@
+package kusto
+
+import (
+	"fmt"
+	"github.com/Azure/azure-kusto-go/kusto/internal/version"
+	"github.com/Azure/azure-kusto-go/kusto/utils"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/samber/lo"
+)
+
+type ClientDetails struct {
+	// applicationForTracing is the name of the application that is using the client.
+	applicationForTracing string
+	// userNameForTracing is the name of the user that is using the client.
+	userNameForTracing string
+	// clientVersionForTracing is the version of the client.
+	clientVersionForTracing string
+}
+
+func NewClientDetails(applicationForTracing string, userNameForTracing string) *ClientDetails {
+	return &ClientDetails{applicationForTracing: applicationForTracing, userNameForTracing: userNameForTracing}
+}
+
+type StringPair struct {
+	Key   string
+	Value string
+}
+
+const NONE = "[none]"
+
+var defaultTracingValuesOnce = utils.NewOnceWithInit[ClientDetails](func() (ClientDetails, error) {
+	return ClientDetails{
+		applicationForTracing:   filepath.Base(os.Args[0]),
+		userNameForTracing:      getOsUser(),
+		clientVersionForTracing: buildHeaderFormat(StringPair{Key: "Kusto.Go.Client", Value: version.Kusto}, StringPair{Key: "Runtime.Go", Value: runtime.Version()}),
+	}, nil
+})
+
+func getOsUser() string {
+	var final string
+	current, err := user.Current()
+	if err == nil && current.Username != "" {
+		final = current.Username
+	} else {
+		// get from env and try domain too
+		final = os.Getenv("USERNAME")
+		domain := os.Getenv("USERDOMAIN")
+		if !isEmpty(domain) && !isEmpty(final) {
+			final = domain + "\\" + final
+		}
+	}
+
+	if isEmpty(final) {
+		final = NONE
+	}
+
+	return final
+}
+
+var escapeRegex = regexp.MustCompile("[\\r\\n\\s{}|]+")
```

Comment:
What's the Go best practice on this?
Should also consts be grouped some place or is it ok to spread them around?

Reply:
I'll put it with the other consts on the top.